diff --git a/source/cbz/mucbz.c b/source/cbz/mucbz.c
index f018d1c..ebaf743 100644
--- a/source/cbz/mucbz.c
+++ b/source/cbz/mucbz.c
@@ -116,8 +116,17 @@ cbz_read_zip_entry(cbz_document *doc, int offset, int *sizep)
 
 	if (method == 0)
 	{
-		*sizep = usize;
-		return cdata;
+		fz_buffer *buf;
+		fz_stream *stm;
+		unsigned char *udata;
+
+		cdata[csize] = 0;
+		stm = fz_open_file (ctx, (char *) cdata);
+		buf = fz_read_all (stm, 0);
+		fz_close (stm);
+		fz_free (ctx, cdata);
+		*sizep = fz_buffer_storage (ctx, buf, &udata);;
+		return udata;
 	}
 
 	if (method == 8)
