diff -ur /home/malc/x/bld/lablgl-1.05/src/Makefile /home/malc/x/inc/lablgl-1.05/src/Makefile
--- /home/malc/x/bld/lablgl-1.05/src/Makefile	2014-08-30 04:40:43.000000000 +0400
+++ /home/malc/x/inc/lablgl-1.05/src/Makefile	2013-09-20 14:50:36.000000000 +0400
@@ -14,12 +14,13 @@
 LIBOBJS =	raw.cmo \
 		gl.cmo		glLight.cmo	glList.cmo	glMap.cmo \
 		glMat.cmo	glMisc.cmo	glPix.cmo	glClear.cmo \
-		glTex.cmo	glDraw.cmo	glFunc.cmo	\
+		glTex.cmo	glDraw.cmo	glFunc.cmo	gluMisc.cmo \
+		gluNurbs.cmo	gluQuadric.cmo	gluTess.cmo	gluMat.cmo \
 		glArray.cmo	glShader.cmo
 MLOBJS = $(LIBOBJS)	togl.cmo
 OPTOBJS = $(LIBOBJS:.cmo=.cmx)
-COBJS =		ml_gl$(XO)	ml_raw$(XO) ml_glarray$(XO) \
-		ml_shader$(XO)
+COBJS =		ml_gl$(XO)	ml_glu$(XO)	ml_raw$(XO) ml_glarray$(XO) \
+		ml_glutess$(XO)	ml_shader$(XO)
 TOGLOBJS =	ml_togl$(XO)	$(TOGLDIR)/togl$(XO)
 
 all: tools
@@ -58,6 +59,9 @@
 gl_tags.c: gl_tags.var
 	$(VAR2SWITCH) -table GL_ < gl_tags.var > $@
 
+glu_tags.c: glu_tags.var
+	$(VAR2SWITCH) GLU_ < glu_tags.var > $@
+
 build.ml: build.ml.in ../Makefile.config
 	sed -e "s|@LABLGL_MLS@|$(LIBOBJS:.cmo=)|" \
 	    -e "s|@TOGL_MLS@|togl|" \
@@ -90,5 +94,6 @@
 
 #dependencies
 ml_gl$(XO): ml_gl.h gl_tags.h gl_tags.c ml_raw.h
+ml_glu$(XO) ml_glutess$(XO) : ml_gl.h ml_glu.h glu_tags.h glu_tags.c
 ml_raw$(XO): raw_tags.h ml_raw.h
 include .depend
